devito/memory.py

```diff
@@ -8,10 +8,9 @@
 import numpy as np
 from sympy import Eq
 
-from devito.dimension import t
 from devito.logger import error
 from devito.tools import convert_dtype_to_ctype
-
+import devito
```

Comment:
Yes, so I think this clearly indicates that we are doing way too much in one super-generic `interfaces` submodule. If we are serious about separating the various data type hierarchies for codegen, arguments and symbolics, we should refactor them into their own submodules, so that the operator only depends on the relevant interfaces, while the functional implementations (`Time/DenseData` et al.) can happily import an operator.

Clearly this is beyond this PR, but we should keep it in mind.